src/main/scala/org/broadinstitute/dsde/workbench/sam/openam/PostgresAccessPolicyDAO.scala

```diff
@@ -362,21 +364,19 @@ class PostgresAccessPolicyDAO(protected val dbRef: DbReference,
     }
   }
 
-  private def insertPolicyActions(actions: Set[ResourceAction], policyId: PolicyPK)(implicit session: DBSession): Int = {
+  private def insertPolicyActions(actions: Set[FullyQualifiedResourceAction], policyId: PolicyPK, descends: Boolean)(implicit session: DBSession): Int = {
     val ra = ResourceActionTable.syntax("ra")
     val rt = ResourceTypeTable.syntax("rt")
-    val r = ResourceTable.syntax("r")
-    val p = PolicyTable.syntax("p")
     val paCol = PolicyActionTable.column
     if (actions.nonEmpty) {
-      val insertQuery = samsqls"""insert into ${PolicyActionTable.table} (${paCol.resourcePolicyId}, ${paCol.resourceActionId})
-            select ${policyId}, ${ra.result.id}
+      val actionsWithResourceTypeSql = actions.map {
+        case FullyQualifiedResourceAction(action, resourceType) => samsqls"(${action}, ${resourceType})"
+      }
+      val insertQuery = samsqls"""insert into ${PolicyActionTable.table} (${paCol.resourcePolicyId}, ${paCol.resourceActionId}, ${paCol.descends})
+            select ${policyId}, ${ra.result.id}, ${descends}
             from ${ResourceActionTable as ra}
             join ${ResourceTypeTable as rt} on ${ra.resourceTypeId} = ${rt.id}
-            join ${ResourceTable as r} on ${r.resourceTypeId} = ${rt.id}
-            join ${PolicyTable as p} on ${p.resourceId} = ${r.id}
-            where ${ra.action} in (${actions})
-            and ${p.id} = ${policyId}"""
```

Comment:
Not exactly. Resource actions are resource type specific. For example, you could have two different `write` resource actions, one for the `workspace` resource type and another for the `billing-project` resource type. When we insert into the `PolicyActionTable`, we need to make sure we are inserting the correct action. 

Before this PR, the correct action was always on the same resource type as the policy. Given that, we would select `${ra.result.id}` based on the resource type of the policy -- that's also why we needed the additional joins to the `ResourceTable` and the `PolicyTable`. 

Now we want to be able to insert actions for arbitrary resource types, not just the resource type of the policy. We are able to do this by passing in the resource type name along with the action and then comparing them to the `(${ra.action}, ${rt.name})` tuples in the conditional statement on l. 379.

So you're right in that we can assume that the `policyId` exists and is valid because we've previously loaded that from Postgres, but that's not why we don't need the conditional anymore